server/src/main/java/org/elasticsearch/common/xcontent/XContentHelper.java

```diff
@@ -453,4 +455,18 @@ public static BytesReference childBytes(XContentParser parser) throws IOExceptio
         builder.copyCurrentStructure(parser);
         return BytesReference.bytes(builder);
     }
+
+    /**
+     * Serialises new XContentType vnd values in a bwc manner
```

Comment:
Can you add an java assert to StreamOutput#.writeEnum to ensure that XContentType is not being serialized there ?